tests/publiccloud/azure_nfs.pm

```diff
@@ -0,0 +1,89 @@
+    # * Download a file, and set the permission to read-only by owner
+    # * Check if the permissiosn are correctly set
+    # * Check if the default user (azureuser) can not read the file
+    # * Check if hard links are working
+    # * Check if soft links are working
+    assert_script_run("cd $share");
+    assert_script_run("curl -v -o Big_Buck_Bunny_8_seconds_bird_clip.ogv " . data_url('Big_Buck_Bunny_8_seconds_bird_clip.ogv'));
+    assert_script_run("sync");
+    assert_script_run("chmod 0400 Big_Buck_Bunny_8_seconds_bird_clip.ogv");
+    assert_script_run("chown root:root Big_Buck_Bunny_8_seconds_bird_clip.ogv");
+    validate_script_output('stat -c "%a" Big_Buck_Bunny_8_seconds_bird_clip.ogv', sub { m/400/ });
+    assert_script_run("! sudo -u azureuser cat Big_Buck_Bunny_8_seconds_bird_clip.ogv >/dev/null");
+    # Check file integrity via md5sums
+    assert_script_run("md5sum Big_Buck_Bunny_8_seconds_bird_clip.ogv > Big_Buck_Bunny_8_seconds_bird_clip.ogv.md5");
+    assert_script_run("mv Big_Buck_Bunny_8_seconds_bird_clip.ogv Big_Buck_Bunny_8_seconds_bird_clip.ogv.orig");
+    # Check hard link
+    my $inode = script_output('stat -c "%i" Big_Buck_Bunny_8_seconds_bird_clip.ogv.orig');    # save inode for later comparison
+    validate_script_output('stat -c "%h" Big_Buck_Bunny_8_seconds_bird_clip.ogv.orig', sub { m/1/ });    # number of hard links must be 1
+    assert_script_run("ln Big_Buck_Bunny_8_seconds_bird_clip.ogv.orig Big_Buck_Bunny_8_seconds_bird_clip.ogv");
+    validate_script_output('stat -c "%h" Big_Buck_Bunny_8_seconds_bird_clip.ogv.orig', sub { m/2/ });    # number of hard links must now be 2
+    validate_script_output('stat -c "%h" Big_Buck_Bunny_8_seconds_bird_clip.ogv', sub { m/2/ });
+    validate_script_output('stat -c "%i" Big_Buck_Bunny_8_seconds_bird_clip.ogv', sub { m/$inode/ });    # compare inode of link
+    assert_script_run("md5sum -c Big_Buck_Bunny_8_seconds_bird_clip.ogv.md5");
+    assert_script_run("rm Big_Buck_Bunny_8_seconds_bird_clip.ogv");
+    assert_script_run("! stat Big_Buck_Bunny_8_seconds_bird_clip.ogv");
+    # Check soft link
+    assert_script_run("ln -s Big_Buck_Bunny_8_seconds_bird_clip.ogv.orig Big_Buck_Bunny_8_seconds_bird_clip.ogv");
+    assert_script_run("md5sum -c Big_Buck_Bunny_8_seconds_bird_clip.ogv.md5");
+    assert_script_run("rm Big_Buck_Bunny_8_seconds_bird_clip.ogv.orig");
+    assert_script_run("stat Big_Buck_Bunny_8_seconds_bird_clip.ogv");
+    assert_script_run("! md5sum -c Big_Buck_Bunny_8_seconds_bird_clip.ogv.md5");    # must fail, because the soft-link is broken now
+    assert_script_run("rm Big_Buck_Bunny_8_seconds_bird_clip.ogv");
+}
+
+sub run {
+    my ($self, $args) = @_;
+    my $instance = $args->{my_instance};
+    my $resource_id = $instance->resource_id;
+    record_info("resource_id", $resource_id);
+
+    # Create mount point and canary to check if remote nfs is mounted
+    assert_script_run('mkdir -p /mount/nfsdata');
```

Comment:
I think it make sense to create variable with mount point value

Reply:
Please elaborate. Do you mean a class variable, a openQA variable or a variable that will be passed to the subroutines? And what value would that bring or which use case are you thinking of? :thinking:

Reply:
just perl variable , use case more clean code where you not repeating same string several times . But if you like to keep it as is it is fine for me